sensor/docs/tutorials/tag_based_pnp_calibrator.md

```diff
@@ -0,0 +1,140 @@
+
+When the user starts the calibration, `rviz` and the `image view` should be displayed like below.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/visualization1.jpg"  alt="visualization1">
+</p>
+
+After the tools detect the LidarTag and AprilTag, it will show the detection markers on the `rviz` and the `image view`. The text in the rviz will also display the current number of pairs of lidar detections and AprilTag detections.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/visualization2.jpg"  alt="visualization2">
+</p>
+
+Once the user gets the converged detection, the user can start moving the tag to another position. Please make sure the moving distance is larger than the `calibration_min_pair_distance` and also make sure the tag is in the FOV of the lidar and camera.
+
+At the end of the calibration, we can get 8 detection pairs which are shown below.
+
+![segment](../images/tag_based_pnp_calibrator/visualization3.jpg)
+
+## Results
+
+After the calibration process is finished, the `sensor_calibration_manager` will display the results in the tf tree and allow the user to save the calibration data to a file.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/menu4.jpg" alt="menu4" width="500">
+</p>
+
+The user can modify the `visualization options` on the right side of the `image view`. To compare the results, please set the `Marker size (m)` to `0.04` and set the `PC subsample factor` to `1`.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/visualization_bar.jpg"  alt="visualization_bar" width="200">
+</p>
+
+After setting the options above, change the `/initial_tf` (in the `visualization options`) to `/current_tf`. By doing this, it is easier to measure the difference after the calibration.
+
+The images below show that with the calibrated transformation, the projected pointcloud aligns better with the image.
+
+<table>
+  <tr>
+    <td><img src="../images/tag_based_pnp_calibrator/init_tf.jpg" alt="init_tf" width = 500px ></td>
+    <td><img src="../images/tag_based_pnp_calibrator/calibrated_tf.jpg" alt="calibrated_tf" width = 500px ></td>
+   </tr>
+   <tr>
+    <td><p style="text-align: center;">Before Calibration.</p></td>
+    <td><p style="text-align: center;">After Calibration.</p></td>
+  </tr>
+</table>
+
+## FAQ
+
+- Why the calibrator doesn't add calibration pairs?
+
+  1. One possible reason is that the current pair is too close to previously collected data. In that case, the current data is not accepted.
+  2. The timestamps of the lidar and camera are not synchronized, this can be checked with `ros2 topic echo [topic_name]`. Setting the parameter `use_receive_time` to `True` might help to solve the issue.
```

Comment:
Solves the issue? Always? It may work but are there any pitfalls?

Reply:
ros2 topic echo will print also the sensor data. --field header.stamp is better

Reply:
Thanks! fixed in https://github.com/tier4/CalibrationTools/pull/164/commits/32dd61431d3beb3d84ecb085cd246042de8efb09